### PR TITLE
Feature/24/block verification method

### PR DESCRIPTION
**Pull request by dct-lau17**

### CODE REVIEW

File: lib/blockchain.rb [hunk 1]
```diff
@@ -19,8 +19,9 @@ def create_transaction(data, time = Time.new)
     unverified_transactions.push(transaction)
   end
 
-  def create_block
-    block = Block.new(unverified_transactions, blocks.last.hash)
+  def create_block(block_class = Block)
```
Comment (archmagos) [inline lib/blockchain.rb]: What's the reason for the block_class change?

File: lib/blockchain.rb [hunk 2]
```diff
@@ -31,4 +32,9 @@ def hash_transaction(data, time)
     sha.update(data['sender'] + data['receiver'] + time.to_s)
     sha.hexdigest
   end
+
+  def invalid?(new_block)
+    current_block = @blocks[@blocks.length - 1]
```
Comment (archmagos) [inline lib/blockchain.rb]: • Blocks shouldn't need an @ as it's covered by an attr_reader (I might be wrong here?)
• blocks.last might be a neater way to get the previous block

File: lib/blockchain.rb (hunk 1 above)
Comment (dct-lau17) [reply lib/blockchain.rb]: Helps me mock, dependency injection

File: lib/blockchain.rb (hunk 2 above)
Comment (dct-lau17) [reply lib/blockchain.rb]: you are right, bad habit :)

File: lib/blockchain.rb [hunk 3]
```diff
@@ -34,7 +34,7 @@ def hash_transaction(data, time)
   end
 
   def invalid?(new_block)
-    current_block = @blocks[@blocks.length - 1]
+    current_block = blocks[blocks.length - 1]
```
Comment (archmagos) [inline lib/blockchain.rb]: Could this just be blocks.last?

File: lib/blockchain.rb (hunk 1 above)
Comment (archmagos) [reply lib/blockchain.rb]: Nice

File: lib/blockchain.rb (hunk 3 above)
Comment (dct-lau17) [reply lib/blockchain.rb]: yes it can XD

**Pull request merged**